test/e2e/secretgencontroller/private_registry_auth_test.go

```diff
@@ -182,42 +141,98 @@ spec:
 		// deletes ServiceAccount before App
 		kubectl.RunWithOpts([]string{"delete", "apps/" + name}, e2e.RunOpts{AllowError: true})
 		kapp.Run([]string{"delete", "-a", name})
+		kapp.Run([]string{"delete", "-a", configName})
 		kapp.Run([]string{"delete", "-a", "secret-export"})
 		kapp.Run([]string{"delete", "-a", registryName, "-n", registryNamespace})
 	}
 	cleanUp()
 	defer cleanUp()
 
+	logger.Section("deploy controller config to skip registry TLS verify", func() {
+		config := `
+apiVersion: v1
+kind: Secret
+metadata:
+  name: kapp-controller-config
+  namespace: kapp-controller
+stringData:
+  dangerousSkipTLSVerify: registry-svc.registry.svc.cluster.local
+`
+		kapp.RunWithOpts([]string{"deploy", "-f", "-", "-a", configName},
+			e2e.RunOpts{StdinReader: strings.NewReader(config)})
+
+		// Since config propagation is async, just wait a little bit
+		time.Sleep(2 * time.Second)
```

Comment:
flaky test detected

Reply:
im open to ideas, but i dont have any.

Reply:
i can't think of anything easy, we'd need some way of turning the config update into a synchronous operation, like an endpoint we could poll or an event emitted, even a line logged. i think i'm fine with ignoring this.